execution/src/transaction.rs

```diff
@@ -18,187 +17,65 @@ use {
-#[serde(untagged)]
-pub enum ExtendedTxEnvelope {
-    Canonical(TxEnvelope),
-    DepositedTx(DepositedTx),
-}
-
-impl ExtendedTxEnvelope {
-    pub fn compute_hash(&self) -> B256 {
-        let mut buf = Vec::with_capacity(self.length());
-        self.encode(&mut buf);
-        keccak256(&mut buf)
-    }
-
-    pub fn sender(&self) -> Option<Address> {
+impl WrapReceipt for OpTxEnvelope {
+    fn wrap_receipt(&self, receipt: Receipt, bloom: Bloom) -> OpReceiptEnvelope {
         match self {
-            Self::Canonical(tx) => tx.recover_signer().ok(),
-            Self::DepositedTx(tx) => Some(tx.from),
-        }
-    }
-
-    /// In case this transaction is a deposit, returns `Some` containing a reference to the
-    /// underlying [`DepositedTx`]. Otherwise, returns `None`.
-    pub fn as_deposited(&self) -> Option<&DepositedTx> {
-        if let Self::DepositedTx(deposited_tx) = self {
-            Some(deposited_tx)
-        } else {
-            None
-        }
-    }
-
-    pub fn wrap_receipt(&self, receipt: Receipt, bloom: Bloom) -> OpReceiptEnvelope {
-        match self {
-            ExtendedTxEnvelope::Canonical(TxEnvelope::Legacy(_)) => {
-                OpReceiptEnvelope::Legacy(ReceiptWithBloom {
-                    receipt,
-                    logs_bloom: bloom,
-                })
-            }
-            ExtendedTxEnvelope::Canonical(TxEnvelope::Eip1559(_)) => {
-                OpReceiptEnvelope::Eip1559(ReceiptWithBloom {
-                    receipt,
-                    logs_bloom: bloom,
-                })
-            }
-            ExtendedTxEnvelope::Canonical(TxEnvelope::Eip2930(_)) => {
-                OpReceiptEnvelope::Eip2930(ReceiptWithBloom {
-                    receipt,
-                    logs_bloom: bloom,
-                })
-            }
-            ExtendedTxEnvelope::DepositedTx(_) => {
+            Self::Legacy(_) => OpReceiptEnvelope::Legacy(ReceiptWithBloom {
+                receipt,
+                logs_bloom: bloom,
+            }),
+            Self::Eip1559(_) => OpReceiptEnvelope::Eip1559(ReceiptWithBloom {
+                receipt,
+                logs_bloom: bloom,
+            }),
+            Self::Eip2930(_) => OpReceiptEnvelope::Eip2930(ReceiptWithBloom {
+                receipt,
+                logs_bloom: bloom,
+            }),
+            Self::Deposit(dep) => {
                 OpReceiptEnvelope::Deposit(OpDepositReceiptWithBloom {
                     receipt: OpDepositReceipt {
                         inner: receipt,
-                        // TODO: what are these fields supposed to be?
-                        deposit_nonce: None,
-                        deposit_receipt_version: None,
+                        // Per OP stack spec <https://specs.optimism.io/protocol/deposits.html#execution>
+                        deposit_nonce: Some(dep.nonce()),
+                        deposit_receipt_version: Some(1),
```

Comment:
nit: we can make the version number a constant: `const DEPOSIT_RECEIPT_VERSION: u64 = 1;`